variable.c

```diff
@@ -72,15 +74,23 @@ struct ivar_update {
 };
 
 void
-Init_var_tables(void)
+Init_Variable(void)
 {
     rb_global_tbl = rb_id_table_create(0);
     generic_iv_tbl_ = st_init_numtable();
-    autoload = rb_intern_const("__autoload__");
+
     /* __classpath__: fully qualified class path */
     classpath = rb_intern_const("__classpath__");
     /* __tmp_classpath__: temporary class path which contains anonymous names */
     tmp_classpath = rb_intern_const("__tmp_classpath__");
+}
+
+void
+Init_Autoload(void)
+{
+    autoload = rb_intern_const("__autoload__");
+
+    rb_eAutoloadError = rb_define_class("AutoloadError", rb_eStandardError);
```

Comment:
I think this will break a compatibility, `rescue NameError` can't catch this.

Reply:
https://github.com/fxn/zeitwerk/pull/95 @byroot specifically asked for a separate exception class, but I don't have a strong opinion, I'd be happy with NameError, but I think there is a difference between autoload failed to define the name or the constant doesn't exist.

Reply:
What about `rb_eAutoloadError = rb_define_class("AutoloadError", rb_eNameError);`

Reply:
I tried it but it's still incompatible with some specs. I think having a specific message is good enough for now.

Reply:
@ioquatix I don't follow, that was a Zeitwerk concern (that we handled), I didn't request any Ruby change. The current `NameError` when the constant is still not defined after successfully loading the file is fine by me.